op-node/node/node.go

```diff
@@ -124,6 +124,16 @@ func (n *OpNode) initL1(ctx context.Context, cfg *Config) error {
 		return fmt.Errorf("failed to create L1 source: %w", err)
 	}
 
+	// Validate the L1 Client Chain ID
+	if err := cfg.Rollup.CheckL1ChainID(ctx, n.l1Source); err != nil {
+		n.log.Error("failed to verify L1 RPC chain id", "err", err)
```

Comment:
I think this should return the error instead of logging it to prevent the node from starting